src/user/views/author_views.py

```diff
@@ -0,0 +1,976 @@
+    SIFT_WEBHOOK_SECRET_KEY,
+)
+from researchhub_comment.models import RhCommentModel
+from researchhub_document.related_models.researchhub_post_model import ResearchhubPost
+from researchhub_document.serializers import DynamicPostSerializer
+from review.models.review_model import Review
+from user.filters import AuthorFilter, UserFilter
+from user.models import Author, Follow, Major, University, User, UserApiToken
+from user.permissions import (
+    Censor,
+    DeleteAuthorPermission,
+    DeleteUserPermission,
+    HasVerificationPermission,
+    RequestorIsOwnUser,
+    UpdateAuthor,
+)
+from user.serializers import (
+    AuthorEditableSerializer,
+    AuthorSerializer,
+    DynamicAuthorProfileSerializer,
+    DynamicUserSerializer,
+    MajorSerializer,
+    UniversitySerializer,
+    UserActions,
+    UserEditableSerializer,
+    UserSerializer,
+)
+from user.tasks import handle_spam_user_task, reinstate_user_task
+from user.utils import calculate_show_referral, reset_latest_acitvity_cache
+from utils.http import POST, RequestMethods
+from utils.openalex import OpenAlex
+from utils.permissions import CreateOrUpdateIfAllowed
+from utils.sentry import log_error, log_info
+from utils.throttles import THROTTLE_CLASSES
+
+
+class AuthorViewSet(viewsets.ModelViewSet):
+    queryset = Author.objects.all()
+    serializer_class = AuthorSerializer
+    filter_backends = (SearchFilter, DjangoFilterBackend, OrderingFilter)
+    filter_class = AuthorFilter
+    search_fields = ("first_name", "last_name")
+    permission_classes = [
+        (IsAuthenticatedOrReadOnly & UpdateAuthor & CreateOrUpdateIfAllowed)
+        | DeleteAuthorPermission
+    ]
+    throttle_classes = THROTTLE_CLASSES
+
+    def create(self, request, *args, **kwargs):
+        """Override to use an editable serializer."""
+        serializer = AuthorEditableSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(
+            serializer.data, status=status.HTTP_201_CREATED, headers=headers
+        )
+
+    def update(self, request, *args, **kwargs):
+        """Override to use an editable serializer."""
+        partial = kwargs.pop("partial", False)
+        instance = self.get_object()
+        serializer = AuthorEditableSerializer(
+            instance, data=request.data, partial=partial
+        )
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+
+        if getattr(instance, "_prefetched_objects_cache", None):
+            # If 'prefetch_related' has been applied to a queryset, we need to
+            # forcibly invalidate the prefetch cache on the instance.
+            instance._prefetched_objects_cache = {}
+
+        return Response(serializer.data)
+
+    @action(detail=True, methods=["get"], permission_classes=[AllowAny])
+    def profile(self, request, pk=None):
```

Comment:
This method is the only change here. This contents of this file was extracted from `user_views.py`